File: R-2.12.1/interpreter/testEnv.R

TestEnv <- function()
{

 this <- list()

 countOk <- 0
 countFail <- 0
 countLatest <- 0

 this$getOk <- function(){ countOk}

 this$test <- function ( b)
 {
  if( b)
  {
   countOk <<- countOk + 1;
  }
  else
  {
   countFail <<- countFail + 1;
   print( 'test failed')
  }
 }

 this$checkComplainAndAdjustExpected <- function( count)
 {
  countTests <- countOk + countFail
  countTestsLocal <- countTests - countLatest

  countLatest <<- countTests

  if( count == countTestsLocal)
  {
   TRUE
  }
  else
  {
   print( sprintf( "Expected Test Count (from last checkpoint to here) doesn't match: countTestsLocal: %d count: %d", countTestsLocal, count))
   FALSE
  }
  unused <- 9
 }

 this$result <- function()
 {
  sprintf( "tests: ok: %d fail: %d countLatest: %d", countOk, countFail, countLatest);
 }

 this
}

File: ocaml-4.01.0/testenv.ml
module TestEnv=
 struct

  let countOk= ref 0 and countFail= ref 0 and countLatest= ref 0

  let test b= 
   if b 
   then
    (
     countOk:= !countOk+ 1;
     true
    )
   else 
    (
     Printf.printf "test failed\n";
     countFail:= !countFail+ 1; 
     false
    )

  let checkComplainAndAdjustExpected count=
   let countTests= !countOk + !countFail in
    let countTestsLocal= countTests - !countLatest in
     (
      countLatest:= countTests;
      if count == countTestsLocal
      then true
      else 
       (
        Printf.printf "Expected Test Count (from last checkpoint to here) doesn't match: countTestsLocal: %d count: %d\n" countTestsLocal count;
        false 
       )
     )

  let result ()=
   Printf.sprintf "tests: ok: %d fail: %d countLatest: %d" !countOk !countFail !countLatest

end;;
